src/Draco.Compiler.Benchmarks/SyntaxBenchmarks.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Generic;
+using BenchmarkDotNet.Attributes;
+using Draco.Compiler.Api.Syntax;
+using Draco.Compiler.Internal.Syntax;
+using SyntaxToken = Draco.Compiler.Internal.Syntax.SyntaxToken;
+
+namespace Draco.Compiler.Benchmarks;
+
+public class SyntaxBenchmarks : FolderBenchmarkBase
+{
+    private SyntaxToken[] tokens = null!;
+
+    public SyntaxBenchmarks()
+        : base("syntax")
+    {
+    }
+
+    [GlobalSetup]
+    public void Setup()
+    {
+        this.tokens = Lex(this.Input);
+    }
+
+    [Benchmark]
+    public void Lex()
+    {
+        var sourceReader = SourceReader.From(this.Input.Code);
+        var syntaxDiagnostics = new SyntaxDiagnosticTable();
+        var lexer = new Lexer(sourceReader, syntaxDiagnostics);
```

Comment:
Having allocations reduce the accuracy of the benchmark, if we don't really need to measure the creation of these objects, I would also put it in init.
But I'm not really sure here and I would ask in #lowlevel.
Same things for the benchmark below.